warehouse/forklift/legacy.py

```diff
@@ -815,6 +815,10 @@ def file_upload(request):
             )
             .one()
         )
+
+        # Update name if it differs but is still equivalent
+        if project.name != form.name.data:
+            project.name = form.name.data
```

Comment:
This needs to be moved after the point where we check if the user has permission to upload the project. As it is now, any user would be able to rename any project.